src/java/org/apache/cassandra/index/sai/disk/v1/trie/LiteralIndexWriter.java

```diff
@@ -40,72 +40,59 @@
  * Builds an on-disk inverted index structure: terms dictionary and postings lists.
  */
 @NotThreadSafe
-public class LiteralIndexWriter implements Closeable
+public class LiteralIndexWriter implements SegmentWriter
```

Comment:
Remove `Closeable` and make the `LiteralIndexWriter` operate similarly to the `NumericIndexWriter`.  The writers are now opened and closed in the `writeCompleteSegment` method.